### PR TITLE
[book] Witness

### PR DESCRIPTION
**Pull request by damirka**
- adds the Witness page
- fixes typos / pre-2024 syntax in some places
- uses The Move Reference where possible
- cleans up SUMMARY
- minor formatting
- removes old and unused pages / source files

// it would be easier to review before I moved the code out of the .md file
// we should consider adding a dev-build on an open PR...

### CODE REVIEW

File: book/src/programmability/witness-pattern.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+`new_coin` function has to be created and consumed _statically_ - by implementing a function in the
+module that owns the `T`. So, if the type `MyCoin` is defined in the `my_coin` module, only this
+module can create a new `Coin` with `T` being `MyCoin`.
+
+```move
+module book::my_coin {
+    use book::simple_coin::{Self, Coin};
+
+    /// The Witness type for the Coin.
+    public struct MyCoin has drop {}
```
[the remark this answers is not in the document]
Comment (healthyred) [reply book/src/programmability/witness-pattern.md:95]: or is it that any struct implemented in this fashion is considered the witness

File: book/src/programmability/witness-pattern.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  function.
+
+## Summary
+
+- Witness is a type with the `drop` ability, used to instantiate generic types with a concrete type
+  parameter.
+- It acts as a proof of ownership of a type and may be used for type-based authorization.
+- Witness can be also used to create abstract class-like behaviors in Move.
+- Transferable Witness is a variation of the Witness that has the `store` ability, allowing it to be
+  stored in objects. It may be useful in certain applications, but not encouraged.
```
Comment (healthyred) [inline book/src/programmability/witness-pattern.md:266]: If the holder of smart contract gates the Transferable Witness with an `AdminCap`, then would this be considered safe as no one else can access nor modify the Transferable Witness?

File: book/src/programmability/witness-pattern.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+module, but rather serves as an initialization value for generic types. For example, if there is a
+generic type `Custom<T>`, where `T` is a type parameter, a witness may be used to instantiate it,
+while also ensuring that no other module can create a `Custom<T>` with the same `T`.
+
+```move
+{{#include ../../../packages/samples/sources/programmability/witness-pattern.move:definition}}
+```
+
+## Use Cases
+
```
Comment (healthyred) [inline book/src/programmability/witness-pattern.md:20]: Another case similar to Proof of Ownership, is combining the witness with the hot potato pattern where you can return a witness in a function, and write a proof function that one of your functions had to have been called.

Comment (thounyy) [inline book/src/programmability/witness-pattern.md:20]: the current use cases' definitions are a bit intertwined. (e.g. Authorization is a an example of why to use a witness as proof of ownership)

**Comment by damirka** [comment]
Thank you @amnn @thounyy and @healthyred! To ship other changes, I'm disabling this page for now and merging.

**Pull request merged**